interchaintest/ibc_client_expire_substitute_test.go

```diff
@@ -0,0 +1,243 @@
+package interchaintest_test
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	"github.com/strangelove-ventures/interchaintest/v3"
+	"github.com/strangelove-ventures/interchaintest/v3/chain/cosmos"
+	"github.com/strangelove-ventures/interchaintest/v3/ibc"
+	"github.com/strangelove-ventures/interchaintest/v3/testreporter"
+	"github.com/strangelove-ventures/interchaintest/v3/testutil"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap/zaptest"
+)
+
+// run `make local-image`to rebuild updated binary before running test
+func TestClientSubstitution(t *testing.T) {
+	if testing.Short() {
+		t.Skip()
+	}
+
+	t.Parallel()
+
+	ctx := context.Background()
+
+	rep := testreporter.NewNopReporter()
+	eRep := rep.RelayerExecReporter(t)
+
+	client, network := interchaintest.DockerSetup(t)
+
+	var (
+		noble                *cosmos.CosmosChain
+		roles                NobleRoles
+		roles2               NobleRoles
+		paramauthorityWallet ibc.Wallet
+	)
+
+	chainCfg := ibc.ChainConfig{
+		Type:           "cosmos",
+		Name:           "noble",
+		ChainID:        "noble-1",
+		Bin:            "nobled",
+		Denom:          "token",
+		Bech32Prefix:   "noble",
+		CoinType:       "118",
+		GasPrices:      "0.0token",
+		GasAdjustment:  1.1,
+		TrustingPeriod: "504h",
+		NoHostMount:    false,
+		Images:         nobleImageInfo,
+		EncodingConfig: NobleEncoding(),
+		PreGenesis: func(cc ibc.ChainConfig) (err error) {
+			val := noble.Validators[0]
+			err = createTokenfactoryRoles(ctx, &roles, denomMetadataRupee, val, false)
```

Comment:
nit: `minSetup` can be `true`

```suggestion
			err = createTokenfactoryRoles(ctx, &roles, denomMetadataRupee, val, true)
```